src/Functions/lgamma.cpp

```diff
@@ -16,6 +16,9 @@ namespace
 /// Use wrapper and use lgamma_r version because std::lgamma is not threadsafe.
 Float64 lgamma_wrapper(Float64 arg)
 {
+    if (arg < 0 && arg == std::floor(arg))
```

Comment:
May I know why `arg == std::floor(arg)`?

Reply:
A number is an integer if and only if it is equal to its floor value. This works for all double values, even for large numbers that don’t fit into int64.